dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataManager.kt

```diff
@@ -53,58 +63,152 @@ class DataManager(
+     * relevant message queues
      * @param storableDataSet contains all the inputs needed by Dataland
      * @return ID of the newly stored data in the data store
      */
     @Transactional
-    fun addDataSet(storableDataSet: StorableDataSet, correlationId: String): String {
+    fun addDataSetToTemporaryStorageAndSendMessage(storableDataSet: StorableDataSet, correlationId: String):
+        String {
         val company = companyManager.getCompanyById(storableDataSet.companyId)
         logger.info(
             "Sending StorableDataSet of type ${storableDataSet.dataType} for company ID " +
                 "${storableDataSet.companyId}, Company Name ${company.companyName} to storage Interface. " +
                 "Correlation ID: $correlationId",
         )
-        val dataId: String = storeDataSet(storableDataSet, company.companyName, correlationId)
-        metaDataManager.storeDataMetaInformation(
-            dataId,
-            storableDataSet.dataType,
-            storableDataSet.uploaderUserId,
-            storableDataSet.uploadTime,
-            company,
+        val dataId: String = storeDataSetInTemporaryStoreAndSendMessage(
+            storableDataSet, company.companyName, correlationId,
+        )
+        val metaData = DataMetaInformationEntity(
+            dataId, storableDataSet.dataType.toString(),
+            storableDataSet.uploaderUserId, storableDataSet.uploadTime, company, QAStatus.Pending,
         )
+        metaDataManager.storeDataMetaInformation(metaData)
         return dataId
     }
 
-    private fun storeDataSet(
+    /**
+     * Method that listens to the qa_queue and updates the metadata information after successful qa process
+     * @param message is the message delivered on the message queue
+     */
+    @RabbitListener(
+        bindings = [
+            QueueBinding(
+                value = Queue("dataQualityAssuredBackendDataManager"),
+                exchange = Exchange(ExchangeNames.dataQualityAssured, declare = "false"),
+                key = [""],
+            ),
+        ],
+    )
+    fun updateMetaData(message: Message) {
+        val dataId = cloudEventMessageHandler.bodyToString(message)
+        val correlationId = message.messageProperties.headers["cloudEvents:id"].toString()
+        if (!dataId.isNullOrEmpty()) {
+            val metaInformation = metaDataManager.getDataMetaInformationByDataId(dataId)
+            metaInformation.qaStatus = QAStatus.Accepted
+            metaDataManager.storeDataMetaInformation(metaInformation)
+            logger.info(
+                "Received quality assurance for data upload with DataId: $dataId with Correlation Id: $correlationId",
+            )
+        } else {
+            val internalMessage = "Error updating metadata data. Correlation ID: $correlationId"
+            logger.error(internalMessage)
+            throw InternalServerErrorApiException(
```

Comment:
We need to reject messages to avoid infinite loops. And we should introduce a dead letter queue

Reply:
Added deadLetterExchange with a queue, which can collect not processed messages. Also added <AmqpRejectAndDontRequeueException> to the message checks. Still open: Introduce manual acknowledgement and try to make the rabbitlisterners leaner.

Reply:
ok